lib/render/SliceRenderer.cpp

```diff
@@ -178,144 +188,231 @@ int SliceRenderer::_resetBoxCache()
-            else
-                dataValues[index + 1] = 0.f;
-
-            dataValues[index] = varValue;
+    // Define a rectangle that encloses our polygon in 3D space.  We will sample along this
+    // rectangle to generate our 2D texture.
+    std::vector<glm::vec3> rectangle3D = {glm::vec3(), glm::vec3(), glm::vec3(), glm::vec3()};
+    auto                   inverseProjection = [&](float x, float y) {
+        glm::vec3 point;
+        point = _origin + x * _axis1 + y * _axis2;
+        return point;
+    };
+    rectangle3D = {glm::vec3(), glm::vec3(), glm::vec3(), glm::vec3()};
+    rectangle3D[3] = inverseProjection(_rectangle2D[0].x, _rectangle2D[0].y);
+    rectangle3D[0] = inverseProjection(_rectangle2D[1].x, _rectangle2D[0].y);
+    rectangle3D[1] = inverseProjection(_rectangle2D[1].x, _rectangle2D[1].y);
+    rectangle3D[2] = inverseProjection(_rectangle2D[0].x, _rectangle2D[1].y);
+
+    return rectangle3D;
+}
 
-            index += 2;
-            coords[X] += deltas[X];
-        }
-        coords[Z] += deltas[Z];
-    }
+// Huges-Moller algorithm to get an orthogonal vector
+// https://blog.selfshadow.com/2011/10/17/perp-vectors/
+glm::vec3 SliceRenderer::_getOrthogonal(const glm::vec3 u) const
+{
+    glm::vec3 a = abs(u);
+    glm::vec3 v;
+    if (a.x <= a.y && a.x <= a.z)
+        v = glm::vec3(0, -u.z, u.y);
+    else if (a.y <= a.x && a.y <= a.z)
+        v = glm::vec3(-u.z, 0, u.x);
+    else
+        v = glm::vec3(-u.y, u.x, 0);
+    v = glm::normalize(v);
+    return v;
 }
 
-void SliceRenderer::_populateDataYZ(float *dataValues, Grid *grid) const
+void SliceRenderer::_populateData(float *dataValues, Grid *grid) const
 {
-    std::vector<double> deltas = _calculateDeltas();
-    float               varValue, missingValue;
-    std::array<double, 3> coords = {0.0, 0.0, 0.0};
-    coords[X] = _cacheParams.boxMin[X];
-    coords[Y] = _cacheParams.domainMin[Y];
-    coords[Z] = _cacheParams.domainMin[Z];
+    float varValue, missingValue;
 
-    int index = 0;
-    for (int j = 0; j < _textureSideSize; j++) {
-        coords[Y] = _cacheParams.domainMin[Y];
+    auto inverseProjection = [&](float x, float y) {
+        glm::vec3 point;
+        point = _origin + x * _axis1 + y * _axis2;
+        return point;
+    };
+
+    glm::vec2 delta = (_rectangle2D[1] - _rectangle2D[0]);
+    delta.x = delta.x / _xSamples;
+    delta.y = delta.y / _ySamples;
+    glm::vec2 offset = {delta.x / 2., delta.y / 2.};
 
-        for (int i = 0; i < _textureSideSize; i++) {
-            varValue = grid->GetValue(coords);
+    int index = 0;
+    for (int j = 0; j < _ySamples; j++) {
+        for (int i = 0; i < _xSamples; i++) {
+            glm::vec3           samplePoint = inverseProjection(offset.x + _rectangle2D[0].x + i * delta.x, _rectangle2D[0].y + offset.y + j * delta.y);
+            std::vector<double> p = {samplePoint.x, samplePoint.y, samplePoint.z};
+            varValue = grid->GetValue(p);
```

Comment:
Grid::GetValue(const std::vector<double> &coords) is being deprecated. Better to use: Grid::GetValue(const CoordType &coords)

I.e. change `std::vector<double> p` to `CoordType p`

Reply:
Got it.  Fixed.